pkg/controller/endpoints/endpoints_controller_test.go

```diff
@@ -0,0 +1,237 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package endpoints_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	rhjmcv1beta1 "github.com/rh-jmc-team/container-jfr-operator/pkg/apis/rhjmc/v1beta1"
+	"github.com/rh-jmc-team/container-jfr-operator/pkg/controller/endpoints"
+	"github.com/rh-jmc-team/container-jfr-operator/test"
+)
+
+var _ = Describe("EndpointsController", func() {
+	var (
+		objs       []runtime.Object
+		client     client.Client
+		controller *endpoints.ReconcileEndpoints
+	)
+
+	JustBeforeEach(func() {
+		logf.SetLogger(zap.Logger())
+		s := test.NewTestScheme()
+
+		client = fake.NewFakeClientWithScheme(s, objs...)
+		controller = &endpoints.ReconcileEndpoints{
+			Client:     client,
+			Scheme:     s,
+			Reconciler: test.NewTestReconcilerNoServer(client),
+		}
+	})
+
+	AfterEach(func() {
+		objs = nil
+	})
+
+	Describe("reconciling a request", func() {
+		Context("successfully reconcile, without JMX auth", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(), test.NewContainerJFRService(),
+					test.NewTargetPod(), test.NewTestEndpoints(),
+				}
+			})
+			It("should create new flightrecorder", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{}))
+
+				found := &rhjmcv1beta1.FlightRecorder{}
+				err = client.Get(context.Background(), types.NamespacedName{Name: "test-pod", Namespace: "default"}, found)
+				Expect(err).ToNot(HaveOccurred())
+				// compare found to desired spec
+				expected := test.NewFlightRecorderNoJMXAuth()
+				Expect(found.TypeMeta).To(Equal(expected.TypeMeta))
+				Expect(found.ObjectMeta.Name).To(Equal(expected.ObjectMeta.Name))
+				Expect(found.ObjectMeta.Namespace).To(Equal(expected.ObjectMeta.Namespace))
+				Expect(found.ObjectMeta.Labels).To(Equal(expected.ObjectMeta.Labels))
+				Expect(found.ObjectMeta.OwnerReferences).To(Equal(expected.ObjectMeta.OwnerReferences))
+				Expect(found.Spec).To(Equal(expected.Spec))
+			})
+		})
+		Context("successfully reconcile, with JMX auth", func() {
```

Comment:
Let's call this "successfully reconcile Container JFR", since it's a special case only for Container JFR itself.